pkg/server/node.go

```diff
@@ -1103,6 +1113,56 @@ func (n *Node) startComputePeriodicMetrics(stopper *stop.Stopper, interval time.
 	})
 }
 
+// startPeriodicLivenessCompaction starts a loop where it periodically compacts
+// the liveness range.
+func (n *Node) startPeriodicLivenessCompaction(stopper *stop.Stopper, interval time.Duration) {
+	ctx := n.AnnotateCtx(context.Background())
+
+	if interval == 0 {
+		log.Infof(ctx, "periodic liveness compaction disabled")
+		return
+	}
+
+	_ = stopper.RunAsyncTask(ctx, "liveness-compaction", func(ctx context.Context) {
+		ticker := time.NewTicker(interval)
+		defer ticker.Stop()
+		for {
+			select {
+			case <-ticker.C:
+				// Find the liveness replica in order to compact it.
+				_ = n.stores.VisitStores(func(store *kvserver.Store) error {
+					store.VisitReplicas(func(repl *kvserver.Replica) bool {
+						if keys.NodeLivenessSpan.Overlaps(repl.Desc().KeySpan().AsRawSpanWithNoLocals()) {
```

Comment:
[nit] Maybe pull `repl.Desc().KeySpan().AsRawSpanWithNoLocals()` into a local var for use here and below?